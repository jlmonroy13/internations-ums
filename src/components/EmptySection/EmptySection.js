import React from 'react';
import PropTypes from 'prop-types';

function EmptySecton({ type, setMenuState, isMenuOpen }) {
  function onSetMenuState() {
    setMenuState(!isMenuOpen);
  }
  return (
    <div className="details__warning">
      <p className="text--larger flush--bottom">{`You don't have any ${type} yet.`}</p>
      <p className="text--large">Please add one.</p>
      <span className="details__warning-btn icon icon--plus" onClick={onSetMenuState} />
    </div>
  );
}

EmptySecton.propTypes = {
  type: PropTypes.string.isRequired,
  setMenuState: PropTypes.func.isRequired,
  isMenuOpen: PropTypes.bool.isRequired,
};

export default EmptySecton;
