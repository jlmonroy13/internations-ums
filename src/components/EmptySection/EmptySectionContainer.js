import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import EmptySection from './EmptySection';
import { setMenuState } from '../../actions/events';

function mapStateToProps(state) {
  const { events: { isMenuOpen } } = state;
  return { isMenuOpen };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ setMenuState }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(EmptySection);
