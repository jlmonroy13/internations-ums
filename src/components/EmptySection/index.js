import EmptySectionContainer from './EmptySectionContainer';

export default EmptySectionContainer;
