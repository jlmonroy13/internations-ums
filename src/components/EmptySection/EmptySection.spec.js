import React from 'react';
import renderer from 'react-test-renderer';
import EmptySection from './EmptySection';

describe('<EmptySection />', () => {
  it('should render the empty section component', () => {
    const setMenuState = jest.fn();
    const wrapper = renderer.create(
      <EmptySection type="" setMenuState={setMenuState} isMenuOpen={true} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
