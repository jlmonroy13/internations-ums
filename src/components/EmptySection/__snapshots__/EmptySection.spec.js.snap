// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptySection /> should render the empty section component 1`] = `
<div
  className="details__warning"
>
  <p
    className="text--larger flush--bottom"
  >
    You don't have any  yet.
  </p>
  <p
    className="text--large"
  >
    Please add one.
  </p>
  <span
    className="details__warning-btn icon icon--plus"
    onClick={[Function]}
  />
</div>
`;
